mdonatello/mdonatello.py

```diff
@@ -9,3 +9,138 @@
 import base64
 
 
+class MoleculeVisualizer:
+    def __init__(self, ag, show_atom_indices=False, highlight_aromatic=False):
+        self.ag = ag.convert_to("RDKit")
+        self.ag_noh = Chem.RemoveHs(self.ag)
+        AllChem.Compute2DCoords(self.ag_noh)
+        self.molecule_list = ["Molecule"]
+        
+        # Create the dropdown and other widgets
+        self.dropdown = Dropdown(
+            options=self.molecule_list,
+            description="Select molecule:",
+            layout=Layout(width="50%")
+        )
+        self.show_atom_indices_checkbox = Checkbox(value=show_atom_indices, description="Show atom indices")
+        self.highlight_aromatic_checkbox = Checkbox(value=highlight_aromatic, description="Highlight aromatic atoms")
+        self.physiochem_props_checkbox = Checkbox(value=False, description="Show Physiochemical Properties")
+        self.hbond_props_checkbox = Checkbox(value=False, description=self.get_hbond_description())
+        self.save_button = Button(description="Save as PNG")
+        
+        # Save button click event
+        self.save_button.on_click(self.save_selected_molecule)
+        
+        # Display widgets
+        self.output_dropdown = VBox()
+        self.output_dropdown.children = [
+            self.dropdown, self.show_atom_indices_checkbox, self.highlight_aromatic_checkbox,
+            self.physiochem_props_checkbox, self.hbond_props_checkbox
+        ]
+        self.output_molecule = VBox()
+        self.output = VBox()
+        self.output.children = [self.output_molecule, self.save_button]
+        
+        display(self.output_dropdown, self.output)
+        
+        # Update display when dropdown value changes
+        self.update_display()
+        
+        # Link widgets to display update
+        self.dropdown.observe(self.update_display, names="value")
+        self.show_atom_indices_checkbox.observe(self.update_display, names="value")
+        self.highlight_aromatic_checkbox.observe(self.update_display, names="value")
+        self.physiochem_props_checkbox.observe(self.update_display, names="value")
+        self.hbond_props_checkbox.observe(self.update_display, names="value")
+
+    def get_hbond_description(self):
```

Comment:
The `get_hbond_description`? Yes it was twice, since it replicated the same things `display_num_h_acceptors` and `display_num_h_donors` do so I removed it in #3.